src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.producer.BulldozerProducer;
+import core.mate.academy.producer.ExcavatorProducer;
+import core.mate.academy.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> list = new ArrayList<>();
+        if (type.equals(Bulldozer.class)) {
+            list = new ArrayList<>(new BulldozerProducer().get());
+        }
+        if (type.equals(Excavator.class)) {
+            list = new ArrayList<>(new ExcavatorProducer().get());
+        }
+        if (type.equals(Truck.class)) {
+            list = new ArrayList<>(new TruckProducer().get());
+        }
+        return list;
```

Comment:
Consider creating a local variable for the `MachineProducer` implementation instead of creating a new `ArrayList` for each type. This would make your code cleaner and would adhere to the checklist item suggesting the use of a local variable in the `getAll` method implementation.